### PR TITLE
Feat/config validation

### PR DESCRIPTION
**Pull request by Nesopie**
@theanmolsharma Had issue with squashes on there :/. This PR has all the updated changes

### CODE REVIEW

**Comment by theanmolsharma** [comment]
umm, okay. I don't understand what issues could have arisen with squashing.

File: src/configuration.model.ts
Comment (theanmolsharma) [inline src/configuration.model.ts:1]: I think you forgot to address [this](https://github.com/Bitshala-Incubator/silent-pay-indexer/pull/6#discussion_r1581866447) comment from my previous review

**Review by theanmolsharma — CHANGES_REQUESTED** [review]
The changes look solid, few final suggestions before we merge.
1. Please sign all your commits. (2nd commit is not signed)
2. Please rebase on main. To merge, we should have a linear history.
3. Please remove any unnecessary commit descriptions and change the commit messages to the following:
    1. chore: add class-transformer and class-validator as dependencies.
    2. feat: add config validation

File: src/configuration.model.ts
Comment (Nesopie) [reply src/configuration.model.ts:1]: OH, I thought you meant the error I was throwing after validation. I've removed everything now

**Review by theanmolsharma — APPROVED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -779,7 +781,7 @@
       "version": "0.3.9",
       "resolved": "https://registry.npmjs.org/@jridgewell/trace-mapping/-/trace-mapping-0.3.9.tgz",
       "integrity": "sha512-3Belt6tdc8bPgAtbcmdtNJlirVoTmEb5e2gC94PnkwEW9jI6CAHUeoG85tjWP5WquqfavoMtMwiG4P926ZKKuQ==",
-      "dev": true,
+      "devOptional": true,
```
Comment (theanmolsharma) [inline package-lock.json:784]: these changes shouldn't be a part of the changeset

**Review by theanmolsharma — CHANGES_REQUESTED** [review]
To fix this, I think you should delete node_modules and do a fresh install with the following command.
`npm install --legacy-peer-deps`

File: package-lock.json (hunk 1 above)
Comment (Nesopie) [reply package-lock.json:784]: ohhh I was installing with --force, and not --legacy-peer-deps. Will reinstall and push!

**Pull request merged**